modules/@apostrophecms/schema/ui/apos/components/AposArrayEditor.vue

```diff
@@ -353,8 +356,31 @@ export default {
     },
     label(item) {
       let candidate;
+      let choice;
       if (this.field.titleField) {
+
+        // Initial field value
         candidate = get(item, this.field.titleField);
+        const titleField = this.schema.find(field => field.name === this.field.titleField);
+
+        // If field is a select, use the choice label as the slat title
+        if (candidate && titleField.type === 'select') {
+
+          // Choices are a normal, hardcoded array
+          if (Array.isArray(titleField.choices)) {
```

Comment:
There's redundancy here which could be avoided by setting a single `titleFieldChoices` property once in `mounted`, by whichever means is appropriate.